python/torch_mlir/compiler_utils.py

```diff
@@ -133,14 +134,35 @@ def forward(self, *args, **kwargs):
 
     return Wrapper(model)
 
-def prepare_model(model, *model_args, dtype = None, **model_kwargs):
+def map_kwargs_into_args(model, model_args, model_kwargs):
+    """
+    Return new_args so that
+        model(*model_args, **model_kwargs)
+    is equivalent to
+        model(*new_args)
+    """
+    func_signature = inspect.signature(model.forward)
+    if any(v.kind == inspect.Parameter.VAR_KEYWORD
+           for v in func_signature.parameters.values() if v.name in model_kwargs):
+        raise TypeError('Keyword-only arguments are not supported')
```

Comment:
Why is this the case? What happens if all arguments are kwargs?

Reply:
It's when the function declares some arguments as keyword-only, see https://peps.python.org/pep-3102/